pkg/registry/pod/etcd/etcd.go

```diff
@@ -105,6 +105,19 @@ func (r *REST) Get(ctx api.Context, name string) (runtime.Object, error) {
 
 // Create creates a pod based on a specification.
 func (r *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, error) {
+
+	//
+	// A user will specify which PVClaim they want to use in their volume, but Kubelet needs
+	// to query for the claim in a namespace.  Set it here to avoid having to require the user to specify it.
+	// I'm sure this bit has a better home somewhere ...
+	//
+	mypod := obj.(*api.Pod)
+	for _, v := range mypod.Spec.Volumes {
+		if v.Source.PersistentVolumeClaimVolumeSource != nil {
+			v.Source.PersistentVolumeClaimVolumeSource.PersistentVolumeClaimRef.Namespace = mypod.Namespace
```

Comment:
I'll try to circle back to secrets today

On Tue, Mar 3, 2015 at 11:50 AM, Mark Turansky <notifications@github.com>
wrote:

> In pkg/registry/pod/etcd/etcd.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4055#discussion_r25719876>
> :
>
> > @@ -105,6 +105,19 @@ func (r *REST) Get(ctx api.Context, name string) (runtime.Object, error) {
> >
> >  // Create creates a pod based on a specification.
> >  func (r *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, error) {
> > +
> > +	//
> > +	// A user will specify which PVClaim they want to use in their volume, but Kubelet needs
> > +	// to query for the claim in a namespace.  Set it here to avoid having to require the user to specify it.
> > +	// I'm sure this bit has a better home somewhere ...
> > +	//
> > +	mypod := obj.(*api.Pod)
> > +	for _, v := range mypod.Spec.Volumes {
> > +		if v.Source.PersistentVolumeClaimVolumeSource != nil {
> > +			v.Source.PersistentVolumeClaimVolumeSource.PersistentVolumeClaimRef.Namespace = mypod.Namespace
>
> I must have forgotten to remove this piece. I added a defaulting function
> to do the same.
>
> but as @pmorie <https://github.com/pmorie> and @smarterclayton
> <https://github.com/smarterclayton> suggest in one of the Secrets PRs,
> perhaps the NewBuilder method can also take a pod (or namespace). That's
> all Paul and I really need in our volumes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4055/files#r25719876>
> .
>